dht/i_partitioner.cc

```diff
@@ -179,6 +179,20 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_not_empty(const dht::token_range& r) {
+    bool is_bounded = r.start() && !r.start()->value().is_minimum() && r.end() && !r.end()->value().is_maximum();
+    if (is_bounded) {
+        uint64_t start_raw = r.start()->value().raw();
+        uint64_t end_raw = r.end()->value().raw();
+        return end_raw - start_raw >= (!r.start()->is_inclusive() + !r.end()->is_inclusive());
+    } else {
+        const static dht::token_range full_range = dht::token_range::make(
+                range_bound<dht::token>(token::from_int64(std::numeric_limits<int64_t>::min() + 1), true),
+                range_bound<dht::token>(token::from_int64(std::numeric_limits<int64_t>::max()), true));
```

Comment:
Maybe use maximum_token() instead of token::from_int64?

Reply:
That will give different results. We want (int64max, +inf) to be empty. Using maximum_token() (which is after_all_keys) will give us a non-empty intersection (int64max, after_all_keys) here.

Reply:
Ok.